src/api/app/models/channel_binary.rb

```diff
@@ -4,6 +4,14 @@ class ChannelBinary < ActiveRecord::Base
   belongs_to :repository
   belongs_to :architecture
 
+  validate do |channel_binary|
+    if channel_binary.project && channel_binary.repository
+      unless channel_binary.repository.project == channel_binary.project
+        errors.add_to_base("Associated project has to match with repository.project")
```

Comment:
we only edit _channel files via the api, are you sure that this error becomes visible then?

It is usually done by a delayed job, the only way to report errors is errbit atm unfortunatly :/